packages/ui/v2/core/Shell.tsx

```diff
@@ -170,6 +172,21 @@ export default function Shell(props: LayoutProps) {
   useRedirectToLoginIfUnauthenticated(props.isPublic);
   useRedirectToOnboardingIfNeeded();
   useTheme("light");
+  const { session } = useRedirectToLoginIfUnauthenticated(props.isPublic);
+
+  const i18n = useViewerI18n();
+
+  // Don't show any content till translations are loaded.
+  // As they are cached infinitely, this status would be loading just once for the app's lifetime until refresh
+  if (i18n.status === "loading") {
+    return (
+      <div className="absolute z-50 flex h-screen w-full items-center bg-gray-50">
+        <Loader />
+      </div>
+    );
+  }
+
+  if (!session && !props.isPublic) return null;
```

Comment:
Id rather show untranslated string than nothing at all. This will increase the First Contentful Paint and will prevent the empty skeleton static generation. Thoughts? @PeerRich @Jaibles